rag/flow/tests/test_title_chunker_ancestor_context.py

```
@@ -0,0 +1,154 @@
+
+        after = next(
+            (ck for ck in chunks if "body after figure" in ck.get("text", "")),
+            None,
+        )
+        assert after is not None, "no chunk contains 'body after figure'"
+        assert "Chapter 1" in after["text"], f"post-media chunk lost ancestor heading, got: {after['text']!r}"
+
+
+def test_post_media_text_keeps_nested_ancestor_heading():
+    """Two-level hierarchy: body text after an image still under ## 1.1
+    must carry the full path "Chapter 1 › 1.1 Background"."""
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `›` (SINGLE RIGHT-POINTING ANGLE QUOTATION MARK). Did you mean `>` (GREATER-THAN SIGN)?

(RUF002)